### PR TITLE
ci(#1591): add ci for demo

### LINKED ISSUE #1591: ci: add ci for demo
Unit tests were added for the demo folder. We shall add a CI for future example's unit tests.

### PR DESCRIPTION
**Pull request by l-lin**
Close #1591.

### CODE REVIEW

**Review by shanmukhateja — COMMENTED** [review]
Could you remove unsued variable `const self = this;` at the first line of `displayTable`?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -19,11 +22,26 @@ jobs:
       - name: Setup Node
         uses: actions/setup-node@v1
         with:
-          node-version: '12.x'
+          node-version: '15.x'
```
Comment (shanmukhateja) [inline .github/workflows/build.yml:25]: Could you tell me why was this needed? Also, odd numbered versions of NodeJS are not LTS releases.

File: src/angular-datatables.directive.ts [hunk 2]
```diff
@@ -81,53 +81,59 @@ export class DataTableDirective implements OnDestroy, OnInit {
...
-                  // finalize context to be sent to user
-                  const _context = Object.assign({}, context, context?.userData, {
-                    adtData: data
-                  });
-                  const instance = self.vcr.createEmbeddedView(ref, _context);
-                  self.renderer.appendChild(cellFromIndex, instance.rootNodes[0]);
-                });
+              if (resolvedDTOptions.columns) {
+                const columns = resolvedDTOptions.columns;
+                this.applyNgPipeTransform(row, columns);
```
Comment (shanmukhateja) [inline src/angular-datatables.directive.ts:86]: I feel like this should be a separate PR :/

**Review by shanmukhateja — COMMENTED** [review]
feedback provided.

File: .github/workflows/build.yml (hunk 1 above)
Comment (l-lin) [reply .github/workflows/build.yml:25]: My bad. I was using node v15 on my computer, and there was this warning in the [build](https://github.com/l-lin/angular-datatables/runs/3390076388?check_suite_focus=true):

> npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!

I'll update to v16.

I didn't know odd numbered versions were not LTS :/

File: src/angular-datatables.directive.ts (hunk 2 above)
Comment (l-lin) [reply src/angular-datatables.directive.ts:85]: It was just some small changes, I just extracted some codes from the `displayTable` method in order to lessen the its length. I don't think it's worth to have its own PR.

**Comment by shanmukhateja** [comment]
LGTM
